File: TME.Store/TME.Api.Contracts/Models/ApiCategory.cs
using System.Collections.Generic;

namespace TME.Store.Api.Models
{
    public class ApiCategory
    {
        public string Id { get; private set; }
        public string ParentId { get; private set; }
        public int Depth { get; private set; }
        public string Name { get; private set; }
        public int TotalProducts { get; private set; }
        public int SubTreeCount { get; private set; }
        public string Thumbnail { get; private set; }
        public List<ApiCategory> SubTree { get; private set; }

        public ApiCategory(string Id, string ParentId, int Depth, string Name, int TotalProducts, int SubTreeCount, string Thumbnail, List<ApiCategory> SubTree)
        {
            this.Id = Id;
            this.ParentId = ParentId;
            this.Depth = Depth;
            this.Name = Name;
            this.TotalProducts = TotalProducts;
            this.SubTreeCount = SubTreeCount;
            this.Thumbnail = Thumbnail;
            this.SubTree = SubTree;
        }
    }
}

File: tme_store/lib/api/models/api_category.dart
import 'package:json_annotation/json_annotation.dart';

part 'api_category.g.dart';

@JsonSerializable()
class ApiCategory {
  @JsonKey(name: 'Id')
  final String id;

  @JsonKey(name: 'ParentId')
  final String parentId;

  @JsonKey(name: 'Depth')
  final int depth;

  @JsonKey(name: 'Name')
  final String name;

  @JsonKey(name: 'TotalProducts')
  final int totalProducts;

  @JsonKey(name: 'Thumbnail')
  final String thumbnaillUrl;

  @JsonKey(name: 'SubTreeCount')
  final int subTreeCount;

  @JsonKey(name: 'SubTree')
  final List<ApiCategory> subTree;

  ApiCategory(
      {this.id,
      this.parentId,
      this.depth,
      this.name,
      this.totalProducts,
      this.thumbnaillUrl,
      this.subTreeCount,
      this.subTree});

  static ApiCategory fromJson(Map<String, dynamic> json) =>
      _$ApiCategoryFromJson(json);

  Map<String, dynamic> toJson() => _$ApiCategoryToJson(this);
}
